tests/test_aip.py

```diff
@@ -0,0 +1,331 @@
+            pytest.skip("Only for integration tests")
+        return os.environ["INTEGRATION_TEST_BUCKET"]
+
+    @pytest.fixture
+    def integration_prefix(self, request):
+        """Fixture that returns the integration test prefix."""
+        marker = request.node.get_closest_marker("integration")
+        if not marker:
+            pytest.skip("Only for integration tests")
+        return os.environ["INTEGRATION_TEST_PREFIX"]
+
+    @pytest.fixture
+    def setup_s3_aip(self, s3_integration_client, integration_bucket, integration_prefix):
+        """Fixture that uploads a test AIP to S3."""
+        uploaded_aips = []
+
+        def _upload_aip(local_aip_directory):
+            bucket = integration_bucket
+            aip_name = local_aip_directory.split("/")[-1]
+            key = f"{integration_prefix}/{aip_name}"
+            uri = f"s3://{bucket}/{key}"
+
+            local_aip_directory = local_aip_directory.removesuffix("/") + "/"
+            for root, _exit, files in os.walk(local_aip_directory):
+                for filename in files:
+                    local_path = os.path.join(root, filename)
+                    relative_path = os.path.relpath(local_path, local_aip_directory)
+                    s3_key = os.path.join(key, relative_path)
+                    s3_integration_client.upload_file(local_path, bucket, s3_key)
+
+            uploaded_aips.append(uri)
+            return uri
+
+        return _upload_aip
+
+    @pytest.mark.integration
+    def test_aip_validate_success(self, setup_s3_aip, integration_prefix):
+        """Test successful validation of a valid AIP."""
+        s3_uri = setup_s3_aip("tests/fixtures/aips/valid-aip")
+        aip = AIP(s3_uri)
+
+        # mock S3 inventory data as we don't have this for ad-hoc uploaded AIPs
+        with patch.object(aip, "_get_aip_s3_inventory") as mocked_inventory:
+            mocked_inventory.return_value = pd.DataFrame(
+                [
+                    {
+                        "key": f"{integration_prefix}/valid-aip/data/hello.txt",
+                        "checksum_algorithm": "NotHelpful",
+                    }
+                ]
+            )
+            result = aip.validate()
+
+        # Assertions
+        assert result.valid
+        assert result.s3_uri == s3_uri
+        assert result.error is None
+
+    @pytest.mark.integration
+    def test_aip_fails_validate_inventory_data_not_found(
+        self, setup_s3_aip, integration_prefix
+    ):
+        """Test validation failure when inventory data is missing."""
+        s3_uri = setup_s3_aip("tests/fixtures/aips/valid-aip")
+        aip = AIP(s3_uri)
+
+        # no mocking required, fails naturally because no S3 inventory for ad-hoc upload
+        result = aip.validate()
+
+        assert not result.valid
+        assert (
+            f"S3 Inventory data not found for S3 key: '{integration_prefix}/valid-aip'"
+            in result.error
+        )
+
+    @pytest.mark.parametrize(
```

Comment:
This was the final iteration of some exploring.   I like the consolidated nature here, when before it was a handful of distinct tests with duplicating code.

I realize it's a little harder to parse at first -- and open to pushback here on that front -- but maybe worth it in the long run.